indico/web/forms/fields.py

```diff
@@ -351,6 +360,12 @@ def _get_event_person(self, data):
 
     def _serialize_principal(self, principal):
         from indico.modules.events.util import serialize_event_person
+        if principal.id is None:
+            # We created an EventPerson which has not been persisted to the
+            # database. Revert the conversion.
+            principal = self.event_person_conversions[principal]
+            if isinstance(principal, dict):
```

Comment:
Is `not isinstance(principal, dict)` expected? If so, do we really want to return `None` in that case?